Add a declarative shadow DOM example (#31577)
mdn/content: files/en-us/web/api/web_components/using_shadow_dom/index.md

The result looks like this:

{{EmbedLiveSample("Imperatively with JavaScript")}}

### Declaratively with HTML

> Creating a shadow DOM via JavaScript API might be a good option for client-side rendered applications. However, in many cases server-side rendered UI might improve the performance and user experience. In such cases, you can use the {{htmlelement("template")}} element to declaratively define the shadow DOM. The key to this behavior is the `shadowrootmode` attribute, which can be set to either `open` or `closed`, the same values as the `mode` option of {{domxref("Element.attachShadow()", "attachShadow()")}} method.

Reviewer: ```suggestion
Creating a shadow DOM via JavaScript API might be a good option for client-side rendered applications. For other applications, server-side rendered UI might have better performance and, therefore, a better user experience. In such cases, you can use the {{htmlelement("template")}} element to declaratively define the shadow DOM. The key to this behavior is the {{glossary("enumerated")}} `shadowrootmode` attribute, which can be set to either `open` or `closed`, the same values as the `mode` option of {{domxref("Element.attachShadow()", "attachShadow()")}} method.
```
adding enumerated because most people tend to think when an attribute has two values its boolean, with a default of "yes". In this case, omitting the attribute, leaving out a value, or using an invalid value, results in "none", not one of the enumerated values, but does not create the shadow dom.

```html
<div id="host">
  <template shadowrootmode="open">
    <span>I'm in the shadow DOM</span>
  </template>
</div>
```

{{EmbedGHLiveSample("dom-examples/shadow-dom/shadowrootmode/simple.html", "", "")}}

> 

Reviewer: I think we need to add a note about the fact that it's the `shadowrootmode="open"` that makes the contents of the shadow root visible, as the contents of `<template>` are not displayed by default. You do mention above, "the key to this", but i think this fact needs to be reinforced, especially since it isn't yet in all browsers. Maybe:

```suggestion
> **Note:** By default, contents of a `<template>` are not displayed. In this case, because the `shadowrootmode="open"` was included, the shadow root is rendered. In supporting browsers, visible contents within that shadow root are displayed.

```

> {{EmbedGHLiveSample("interactive-examples/live-examples/html-examples/template-shadowrootmode/simple.html", "", "")}}

Reviewer: ```suggestion
{{EmbedGHLiveSample("html-examples/template-shadowrootmode/simple.html", "", "")}}
```
it's not showing. Drop the "interactive-examples/live-examples/" from the repo URL.

This is the format from https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#result_4, which works . That links to https://mdn.github.io/html-examples/custom-select/ 
However, https://mdn.github.io/html-examples/template-shadowrootmode/simple.html is 404'ing, so maybe the repo needs to be pushed?

Reviewer: https://github.com/mdn/interactive-examples/blob/main/CONTRIBUTING.md

Author: I used the dom-examples repository, which makes demos immediately available after the merge.
